src/rust/engine/fs/store/src/snapshot_ops_tests.rs

```diff
@@ -120,6 +120,24 @@ async fn subset_recursive_wildcard() {
     .await
     .unwrap();
   assert_eq!(merged_digest, subset_roland2);
+
+  // ** should not include explicitly excluded files
+  let subset_params3 = make_subset_params(&["!subdir/roland2", "subdir/**"]);
+  let subset_roland3 = store
+    .clone()
+    .subset(merged_digest, subset_params3)
+    .await
+    .unwrap();
+  assert_eq!(subset_roland3, snapshot1.digest);
+
+  // ** should not include explicitly excluded files
+  let subset_params4 = make_subset_params(&["!subdir/roland2", "**"]);
+  let subset_roland4 = store
+    .clone()
+    .subset(merged_digest, subset_params4)
+    .await
+    .unwrap();
+  assert_eq!(subset_roland4, snapshot1.digest);
```

Comment:
This fails for some reason, and I've not figured out yet how to inspect the snapshot to see if it is correct or not..

Reply:
`Snapshot` manually implements `Debug` to hide that information, but it doesn't need to. To fix that, can apply a diff like:
```diff
diff --git a/src/rust/engine/fs/store/src/snapshot.rs b/src/rust/engine/fs/store/src/snapshot.rs
index a94f033e7..1c0c79bf4 100644
--- a/src/rust/engine/fs/store/src/snapshot.rs
+++ b/src/rust/engine/fs/store/src/snapshot.rs
@@ -19,7 +19,7 @@ use itertools::Itertools;
 
 use crate::Store;
 
-#[derive(Eq, Hash, PartialEq)]
+#[derive(Eq, Debug, Hash, PartialEq)]
 pub struct Snapshot {
   pub digest: Digest,
   pub path_stats: Vec<PathStat>,
@@ -305,17 +305,6 @@ impl Snapshot {
   }
 }
 
-impl fmt::Debug for Snapshot {
-  fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-    write!(
-      f,
-      "Snapshot(digest={:?}, entries={})",
-      self.digest,
-      self.path_stats.len()
-    )
-  }
-}
-
 fn paths_of_child_dir(paths: Vec<PathStat>) -> Vec<PathStat> {
   paths
     .into_iter()
```

Then, to dump out the `Snapshot`, you could use:
```rust
// In standard debug format.
println!("snapshot: {:?}", snapshot);`
// In pretty-printed debug format:
println!("snapshot: {:#?}", snapshot);`
```

Reply:
Now this test passes, yay :D